### DISCUSSION TITLE
Transmit BAP's description to BPP so as to render it for BPP agent's reference

### DISCUSSION THREAD

**Discussion by NikithShetty** [beckn/protocol-specifications#365] [Proposals]
### Context
Let's take the mobility use case. In a mobility network, driver is onboarded by a provider who can receive ride request from multiple BAPs. Driver would need to know this information to help with the communication between the driver and the customer before the ride starts.

### Problem
BAP's listing details such as name, logo, description etc, are not sent to BPP during the transaction. Only BPPs details are sent to BAP using the `provider.description` object.

### Proposed Solution
Add `application.description` object with similar schema as `provider.description` as part of `/search` request.

#### /search req:
`message.intent.application.description`

**Comment by ravi-prakash-v** [comment]
@NikithShetty There are many approaches to this. Before we narrow down on a solution, let us identify the core business problem statement. 

The first question is "why"; Why will the driver need to know which platform the ride request is coming from? For him a ride coming from say Google is the same as a ride coming from PayTM.  Why should a BPP agent need BAP platform context?

**Comment by NikithShetty** [comment]
@ravi-prakash-v One use case where this is required is for driver <-> rider connect via call, before ride start.

Example: rider calls up driver, to enquire where they are, and says "Hello, is this NammaYatri?", and the driver is onboarded by "Mobility Cabs" BPP, the driver might be confused as to who is calling and why.

We show the provider information to the rider, however the driver is oblivious of the BAP.

**Reply by ravi-prakash-v** [reply]
That is an user experience issue. If the rider is say, on a Google Maps BAP, is he going to say, _"Hey is this Google Maps?"_. 

The BAPs UI must render the experience in such a way that the riders do not get confused who they are talking to. If the Namma Yatri BAP clearly shows '**"Bob from Mobility Cabs"** has been assigned as your driver', then it will not lead to any confusion on the user's part. 

I would NOT recommend informing the driver about which platform the request is coming from. BAPs are **experience providers**. They must be _completely_ provider agnostic.  When designing a BAP, the designer should strive to keep the provider context _completely out of scope_ and focus only on the _use case_ it supports.
